src/test/java/jenkins/scm/impl/SingleSCMSourceTest.java

```diff
@@ -220,6 +224,14 @@ public void getSCMDescriptors() throws Exception {
                 (Matcher) hasItem(instanceOf(MockSCM.DescriptorImpl.class)));
     }
 
+    @Issue("JENKINS-52964")
+    @Test
+    public void filesystem_supports_false_by_default_for_descriptor() {
```

Comment:
There's testing of the positive case down in https://github.com/jenkinsci/git-plugin/pull/623 if that helps - otherwise, lemme know what other scenarios you'd like to see tests for.

Reply:
Scm-api should have positive and negative test cases. Relying on a downstream repo will not prevent broken releases.

Also in case I forgot, MINOR version bump

Reply:
Gotcha, will do.

And I already did bump the version to 2.3.0 after you brought it up before. =)

Reply:
> And I already did bump the version to 2.3.0 after you brought it up before. =)

ack. it's hard to review on a phone!

Reply:
Fun, this is going to require a couple mock `SCMFileSystem` implementations - I'll try to get that done tomorrow.

Reply:
Just extend the mock with some other implementations, registered with `@TestExtension` and then remove the mock we have's builder from the extension list when you want to test